internal/memo/memStorage.go

```diff
@@ -0,0 +1,165 @@
+package memo
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"internal/dbaser"
+	"sync"
+
+	"os"
+)
+
+type Metrics struct {
+	ID    string   `json:"id"`              // имя метрики
+	MType string   `json:"type"`            // параметр, принимающий значение gauge или counter
+	Delta *int64   `json:"delta,omitempty"` // значение метрики в случае передачи counter
+	Value *float64 `json:"value,omitempty"` // значение метрики в случае передачи gauge
+}
+
+type Gauge float64
+type Counter int64
+type gauge = Gauge
+type counter = Counter
+type MemStorage struct {
+	Gaugemetr map[string]gauge
+	Countmetr map[string]counter
+	Mutter    sync.RWMutex
```

Comment:
Вся механика блокировок должна быть скрыта внутри стораджа.
```suggestion
	mutter    sync.RWMutex
```